backend/controller/controller.go

```diff
@@ -128,16 +129,23 @@ func Start(ctx context.Context, config Config, runnerScaling scaling.RunnerScali
 
 	console := NewConsoleService(dal)
 
-	ingressHandler := http.StripPrefix("/ingress", svc)
+	ingressHandler := http.Handler(svc)
 	if len(config.AllowOrigins) > 0 {
 		ingressHandler = cors.Middleware(slices.Map(config.AllowOrigins, func(u *url.URL) string { return u.String() }), ingressHandler)
 	}
 
+	// Start the separate HTTP server for ingress
+	go func() {
+		err := rpc.ServeHTTP(ctx, config.IngressBind, ingressHandler)
+		if err != nil {
+			logger.Errorf(err, "Ingress server failed")
```

Comment:
Maybe we want to return an error here instead of logging? I wasn't 100% sure since I wanted to keep the current setup of returning the `rpc.Serve` result.

Reply:
We definitely do want to return an error. Just put both into an errgroup an return eg.Wait()

Reply:
I really don't think we should be using an `rpc.Server` for this, it has a whole lot of machinery that we just don't want for normal HTTP serving. Better to create a new package `internal/http` (or similar).